index.d.ts

```diff
@@ -80,8 +118,11 @@ declare class GenericSpan implements Taggable {
   // timestamp, ended, id, traceId, parentId, sampled, duration()
 
   type: string;
+  subtype: string;
+  action: string;
   traceparent: string;
 
+  setType (type: string, subtype: string, action: string): void;
```

Comment:
I think we need to allow for the following ways of calling `setType`:

```js
span.setType('a', 'b', 'c')
span.setType('a', 'b', null)
span.setType('a', null, null)
span.setType('a', 'b')
span.setType('a', null)
span.setType('a')
```

The following TypeScript Typings should allow for that and nothing else:

```suggestion
  setType (type: string, subtype?: string | null): void;
  setType (type: string, subtype: string, action?: string | null): void;
  setType (type: string, subtype?: null, action?: null): void;
```

And we should add tests for this in `test/types/index.ts` as well to test that this all works as expected 😄

Reply:
I just made every field optional _and_ `string | null`, so it can accept _any_ combination of strings, nulls or undefined.